### PR TITLE
[SCB-1065] when request not contain traceId,should use provider's invocation's traceId

### PR DESCRIPTION
**Pull request by weichao666**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ protected void setContext() throws Exception {
     @SuppressWarnings("unchecked")
     Map<String, String> cseContext =
         JsonUtils.readValue(strCseContext.getBytes(StandardCharsets.UTF_8), Map.class);
-    invocation.setContext(cseContext);
+    invocation.addContext(cseContext);
```
Comment (wujimin) [inline common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java]: highway have the same problem

we can add a new method in invocation: mergeContext
1.if new context have more items, then addAll to new context, and replace old context
2.if new context have less items, then allAll to old context directly.

File: swagger/swagger-invocation/invocation-core/src/main/java/org/apache/servicecomb/swagger/invocation/context/InvocationContext.java [hunk 2]
```diff
@@ -68,6 +68,22 @@ public void addContext(Map<String, String> otherContext) {
...
+  public void mergeContext(InvocationContext otherContext) {
+    mergeContext(otherContext.getContext());
+  }
+
+  public void mergeContext(Map<String, String> otherContext) {
+    if (otherContext == null) {
+      return;
+    }
+    if (otherContext.size() > context.size()) {
+      otherContext.putAll(context);
```
Comment (wujimin) [inline swagger/swagger-invocation/invocation-core/src/main/java/org/apache/servicecomb/swagger/invocation/context/InvocationContext.java]: 1.loop context and otherContext.putIfAbsent
2.still did not process highway transport?

Comment (jeho0815) [reply swagger/swagger-invocation/invocation-core/src/main/java/org/apache/servicecomb/swagger/invocation/context/InvocationContext.java]: > 1.loop context and otherContext.putIfAbsent
> 2.still did not process highway transport?

new traceid created in invocation.onStart, then merge context. if only merge absent keys, the traceid is the new one.
can we put the merge context before invocation.onStart?

File: common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java (hunk 1 above)
Comment (weichao666) [reply common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java]: done

**Pull request merged**